### PR TITLE
Major refactor: server/db, pg-promise, saveDraft (odp) using transaction

### LINKED ISSUE #806: ODPService/repositories with transactions
- [ ] Introduce pg-promise and add DB object 
- [ ] Refactor ODPService so that its methods call other repositories (do not import a different repository from another repository)
- [ ] Transactions always start in Service and are passed down to repositories

### PR DESCRIPTION
**Pull request by sorja**
Title

Part of #806

### CODE REVIEW

**Comment by minotogna** [comment]
does it resolves all the issue?

File: server/db/db.ts [hunk 1]
```diff
@@ -1,60 +1,54 @@
-// Mostly copied from an example here: https://github.com/brianc/node-postgres
-import * as pg from 'pg'
+import 'dotenv/config'
```
Comment (minotogna) [inline server/db/db.ts:1]: do we need this here? we already have it in server/index.ts

File: server/repository/odp/create.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { CountryIso } from '@core/country'
+import { BaseProtocol, DB } from '@server/db/db'
```
Comment (minotogna) [inline server/repository/odp/create.ts:2]: { BaseProtocol, DB} must be imported from '@server/db'

File: server/repository/odp/markAsActual.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { insertAudit } from '@server/repository/audit/auditRepository'
+import Promise from 'bluebird'
```
Comment (minotogna) [inline server/repository/odp/markAsActual.ts:2]: we should get rid of Bluebird.

File: server/repository/odpClass/wipeNationalClassIssues.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ export const wipeNationalClassIssues = async (client: any, odpId: any, countryIs
     `,
     hasClasses ? [countryIso, 'odp', ...classUuids] : [countryIso, 'odp']
   )
-  const issueIds = res.rows.map((r: any) => r.issue_id)
+  const issueIds = res.map((r: any) => r.issue_id)
```
Comment (minotogna) [inline server/repository/odpClass/wipeNationalClassIssues.ts:27]: can we get rid of Ramda (code above)?

File: server/repository/odpDraft/odpDraftRepository.ts [hunk 5]
```diff
@@ -1,14 +1,12 @@
 import { OdpClassRepository } from '@server/repository'
-import { insertAudit } from '../audit/auditRepository'
 
 export const getDraftId = async (client: any, odpId: any) => {
```
Comment (minotogna) [inline server/repository/odpDraft/odpDraftRepository.ts:3]: this operates on ODP table. we might not need OdpDraftRepository.
A repository is a one to one correspondence with a db table

File: server/repository/odpDraft/updateOrInsertDraft.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    await updateDraft(client, draft)
+    await OdpClassRepository.wipeNationalClassIssues(client, odpId, countryIso, draft.nationalClasses)
+  } else {
+    await insertDraft(client, countryIso, user, odpId, draft)
+  }
+
+  await insertAudit(client, user.id, 'updateOrInsertDraft', countryIso, 'odp', { odpId })
+
+  return { odpId }
+}
```
Comment (minotogna) [inline server/repository/odpDraft/updateOrInsertDraft.ts:18]: this belong to service

File: server/repository/odpDraft/odpDraftRepository.ts [hunk 7]
```diff
@@ -1,14 +1,12 @@
...
-  const res = await client.query('SELECT draft_id FROM odp WHERE id = $1', [odpId])
-  return res.rows[0].draft_id
+  const [{ draft_id: draftId }] = await client.query('SELECT draft_id FROM odp WHERE id = $1', [odpId])
+  return draftId
 }
 
 export const updateDraft = async (client: any, draft: any) => {
```
Comment (minotogna) [inline server/repository/odpDraft/odpDraftRepository.ts:8]: this belong to service. one repository cannot import another repository

File: server/service/odp/create.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+import { BaseProtocol, DB } from '@server/db/db'
+import { insertAudit } from '@server/repository/audit/auditRepository'
+
+export const create = (options: { countryIso: CountryIso; user: User }, client: BaseProtocol = DB) => {
+  const { countryIso, user } = options
+  return client.tx(async (t) => {
+    const odpId = OdpRepository.create({ countryIso }, t)
+    await insertAudit(t, user.id, 'createOdp', countryIso, 'odp', { odpId })
+    return odpId
+  })
```
Comment (minotogna) [inline server/service/odp/create.ts:13]: transactions must be started in the service. one repository cannot use another repository

**Review by minotogna — CHANGES_REQUESTED** [review]
please see my comments, plus not all the issue is resolved if I'm not wrong.

File: server/repository/odpDraft/updateOrInsertDraft.ts (hunk 6 above)
Comment (sorja) [reply server/repository/odpDraft/updateOrInsertDraft.ts:18]: separate pr

File: server/repository/odpDraft/odpDraftRepository.ts (hunk 7 above)
Comment (sorja) [reply server/repository/odpDraft/odpDraftRepository.ts:8]: separate pr

**Comment by sorja** [comment]
> 
> 
> please see my comments, plus not all the issue is resolved if I'm not wrong.

Part 1 of larger concept, making smaller PRs
Updated description for correspond

File: server/repository/odpDraft/updateOrInsertDraft.ts (hunk 6 above)
Comment (minotogna) [reply server/repository/odpDraft/updateOrInsertDraft.ts]: why?

File: server/repository/odpDraft/odpDraftRepository.ts (hunk 7 above)
Comment (minotogna) [reply server/repository/odpDraft/odpDraftRepository.ts:3]: why?

**Review by minotogna — APPROVED** [review]

**Pull request merged**